### PR TITLE
Add config option to disable Civilians Yelling before they move

### LINKED ISSUE #717: Now that civilian noise alerts have been corrected, remove the broken default behavior
Add the option to disable the default civilian yell in class XGAIBehavior_Civilian.  This default yell behavior interferes with the modified donoisealert() in class X2AIBTDefaultActions.  This is copied from Long War.  

```
function BTRunCompletePreExecute()
{
	local AvailableAction YellAction;
	local int TargetIndex;
	super.BTRunCompletePreExecute();
    
    // LWS Add: Configurable disabling of civilian yell
    if (!class'Helpers_LW'.default.EnableCivilianYellOnPreMove)
        return;

	if( !m_kPlayer.bCiviliansTargetedByAliens )
	{
		// Yell before moving.
		YellAction = FindAbilityByName('Yell');
		if( YellAction.AbilityObjectRef.ObjectID > 0 && YellAction.AvailableCode == 'AA_Success' )
		{
			if( YellAction.AvailableTargets.Length == 0 )
			{
				TargetIndex = INDEX_NONE;
			}
			class'XComGameStateContext_Ability'.static.ActivateAbility(YellAction, TargetIndex);
		}
	}
}
```

### PR DESCRIPTION
**Pull request by Tedster59**
Fixes #717 

This will help mods that adjust AI behavior.  Will allow Yellow Alert 2.0 (and soon LWoTC) to remove an MCO of `XGAIBehavior_Civilian`

### CODE REVIEW

File: X2WOTCCommunityHighlander/Config/XComGame.ini [hunk 1]
```diff
@@ -266,3 +266,7 @@ bEnableVersionDisplay=true
 ;;; HL-Docs: ref:EuroDateOverride
 ; Uncomment the following line to set European date style for INT localization
 ; bForceEuroDateStrings = true
+
+;;; HL-Docs: ref:CiviliansPreMoveYell
+; Uncomment the following line to disable the civilian pre-move yell mechanic
+; bDisableCivilianPreMoveYell = true
```
Comment (Iridar) [inline X2WOTCCommunityHighlander/Config/XComGame.ini:272]: It's not a big deal, but in the future please try to maintain an empty line at the end of all files.

Comment (Tedster59) [reply X2WOTCCommunityHighlander/Config/XComGame.ini]: Added

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/XGAIBehavior_Civilian.uc [hunk 2]
```diff
@@ -129,15 +129,22 @@ function BTRunCompletePreExecute()
 	super.BTRunCompletePreExecute();
 	if( !m_kPlayer.bCiviliansTargetedByAliens )
 	{
-		// Yell before moving.
-		YellAction = FindAbilityByName('Yell');
-		if( YellAction.AbilityObjectRef.ObjectID > 0 && YellAction.AvailableCode == 'AA_Success' )
+
+		/// HL-Docs: feature:CiviliansPreMoveYell; issue:717; tags:
+		/// Add config option to disable the civilian pre-move "Yell", useful for AI modification mods.
+		if(!class'CHHelpers'.bDisableCivilianPreMoveYell)
```
Comment (Iridar) [inline X2WOTCCommunityHighlander/Src/XComGame/Classes/XGAIBehavior_Civilian.uc:135]: This needs to be `class'CHHelpers'.default.bDisableCivilianPreMoveYell`. Please at least compile test all your Highlander submissions.

**Review by Iridar — CHANGES_REQUESTED** [review]

File: X2WOTCCommunityHighlander/Src/XComGame/Classes/XGAIBehavior_Civilian.uc (hunk 2 above)
Comment (Tedster59) [reply X2WOTCCommunityHighlander/Src/XComGame/Classes/XGAIBehavior_Civilian.uc]: Ok, fixed that boneheaded mistake

**Comment by Iridar** [comment]
There was an .x2proj conflict, I tried resolving it via GitHub interface and it just FUBAR'ed the entire PR and I don't how to unfox it. Sorry, but I'm really not in the mod for git bullshit, so I'll just recreate the PR.

**Pull request closed without merging**